### ISSUE TITLE
latest intel caffe + resnet cpp_classification crash

### ISSUE THREAD

**Issue by etaf** [intel/caffe#32]
Hi, I tried to run the caffe example cpp_classification using latest intel caffe + mkl2017, but crashed.
Help!

The error:
>F0106 21:40:14.463059 29978 net.cpp:1227] Check failed: target_blobs.size() == source_layer.blobs_size() (2 vs. 3) Incompatible number of blobs for layer bn_conv1

How I build:
>cmake  -DCPU_ONLY=ON -DUSE_MKL2017_AS_DEFAULT_ENGINE=ON

Here is my configuration:
>system: ubuntu 14.04
intel caffe: commit:  0bc848bd32d17d5f17bfc7e20915e9805fd1f180
mkl: mklml_lnx_2017.0.1.20161005
resnet caffe model:  https://github.com/KaimingHe/deep-residual-networks

Thanks.

**Comment by etaf** [comment]
The same problem in issue: https://github.com/intel/caffe/issues/1  clams it's been fixed, but I check the related commit, It did not.
Are you going to fix this ?
I believe this is an bug.

**Comment by etaf** [comment]
The root cause:
When initialize a network from a prototxt:
For Batch normalization layer, intel caffe will create layer using layer factory.
For mkldnn it will use mkldnn batch normalization layer and for mkl it will use mkl batch normalization.
There is no problem when use caffe engine.
When use mkldnn or mkl2017 as engine, the blob size of batch normalization layer will be reshape to 2, original is 3.
After the network initialized, we start to load the pre-trained mode weights.
For Batch normalization layer, the blob size is 3. 
Since the layer we created has blob size 2, the program crashed.

**Comment by michalkuligowski** [comment]
Thanks for reporting this issue. We were able to reproduce and are working on fixing this error.

**Comment by jczaja** [comment]
@etaf: We released some fixes that should fix your problem. Please sync and test and come back to us with our feedback. Have a nice day!
